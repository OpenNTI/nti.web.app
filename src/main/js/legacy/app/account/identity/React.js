const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
