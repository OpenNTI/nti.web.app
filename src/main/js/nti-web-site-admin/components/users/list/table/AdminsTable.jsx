import React from 'react';
import { scoped } from '@nti/lib-locale';
import { decorate } from '@nti/lib-commons';
import { searchable, contextual } from '@nti/web-search';

import UsersTable from './UsersTable';
import Store from './Store';

const t = scoped('nti-web-site-admin.users.list.table.AdminsTable', {
	administrators: 'Administrators',
	emptyMessage: 'There are no current administrators',
});

class AdminsTable extends React.Component {
	static deriveStateKeyFromProps = () => 'AdminsTable';
	static deriveFilterFromProps = () => 'admin';

	render() {
		return (
			<UsersTable
				filter="admin"
				title={t('administrators')}
				emptyMessage={t('emptyMessage')}
			/>
		);
	}
}

export default decorate(AdminsTable, [
	searchable(),
	contextual(t('administrators')),
	Store.connect({}),
]);
