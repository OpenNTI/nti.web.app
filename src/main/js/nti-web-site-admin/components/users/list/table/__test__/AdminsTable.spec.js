import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import AdminsTable from '../AdminsTable';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = (numberOfUsers) => {
	return {
		getCollection: () => {
			return {
				hasLink: () => true,
				getLink: () => 'mockLink'
			};
		},
		getBatch: () => {
			let Items = [];

			for(let i = 0; i < numberOfUsers; i++) {
				const index = i + 1;

				Items.push({
					Username: 'test' + index,
					getID: () => 'test' + index,
					getCreatedTime: () => new Date('10/30/2017'),
					getLastSeenTime: () => new Date(Date.now() - 1000 * 60 * 60 * 24 * index) // last seen i days ago
				});
			}

			return Promise.resolve({
				Total: numberOfUsers,
				BatchPage: 1,
				Items
			});
		},
		getWorkspace: () => {
			return {
				getLink: () => 'mockLink'
			};
		}
	};
};

const onBefore = (numberOfUsers) => {
	jest.useFakeTimers();
	setupTestClient(getMockService(numberOfUsers));
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe('Site admin user admin list (with no items)', () => {
	beforeEach(() => onBefore(0));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<AdminsTable/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin user admin list (with 5 items)', () => {
	beforeEach(() => onBefore(5));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<AdminsTable/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin user admin list (with 25 items)', () => {
	beforeEach(() => onBefore(25));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<AdminsTable/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
