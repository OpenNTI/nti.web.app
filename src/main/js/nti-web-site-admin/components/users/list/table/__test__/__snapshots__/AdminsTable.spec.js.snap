// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user admin list (with 5 items) Basic render test 1`] = `
<div
  className="users-table-container"
>
  <div>
    <div
      className="header"
    >
      <div
        className="title"
      >
        Administrators
      </div>
    </div>
    <table
      className="nti-generic-table"
    >
      <thead>
        <tr>
          <th
            className=""
          >
            <label
              className="checkbox-component"
            >
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />
              <span
                className="label"
              />
            </label>
          </th>
          <th
            className="nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            className="nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Join Date
              </span>
            </div>
          </th>
          <th
            className="nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Last Active
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              1 Day ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              2 Days ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              3 Days ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              4 Days ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              5 Days ago
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Site admin user admin list (with 25 items) Basic render test 1`] = `
<div
  className="users-table-container"
>
  <div>
    <div
      className="header"
    >
      <div
        className="title"
      >
        Administrators
      </div>
    </div>
    <table
      className="nti-generic-table"
    >
      <thead>
        <tr>
          <th
            className=""
          >
            <label
              className="checkbox-component"
            >
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />
              <span
                className="label"
              />
            </label>
          </th>
          <th
            className="nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            className="nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Join Date
              </span>
            </div>
          </th>
          <th
            className="nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Last Active
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              1 Day ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              2 Days ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              3 Days ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              4 Days ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              5 Days ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              6 Days ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              1 Week ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              1 Week ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              1 Week ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              1 Week ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              1 Week ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              1 Week ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              1 Week ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              2 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              2 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              2 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              2 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              2 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              2 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              2 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              3 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              3 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              3 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              3 Weeks ago
            </div>
          </td>
        </tr>
        <tr>
          <td
            className=""
          >
            <div>
              <label
                className="checkbox-component"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="label"
                />
              </label>
            </div>
          </td>
          <td>
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div>
                <div>
                  <div />
                </div>
              </div>
            </div>
          </td>
          <td>
            <div>
              <time
                className="ntiText"
                dateTime="2017-10-30T00:00:00+00:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td>
            <div>
              3 Weeks ago
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="site-admin-table-pager"
    >
      <div
        className="page selected"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="page"
        onClick={[Function]}
      >
        2
      </div>
    </div>
  </div>
</div>
`;

exports[`Site admin user admin list (with no items) Basic render test 1`] = `
<div
  className="users-table-container"
>
  <div
    className="empty-state"
  >
    There are no current administrators
  </div>
</div>
`;
